core/src/main/kotlin/org/evomaster/core/search/EvaluatedIndividual.kt

```diff
@@ -783,6 +784,11 @@ class EvaluatedIndividual<T>(
         }
         mutatedGenes.addedInitializationGenes.addAll(diff.flatMap { it.seeTopGenes() })
 
+        if (addedInsertions!!.flatten().isEmpty()) {
```

Comment:
@man-zhang what would be the case of having `! diff.isEmpty()` but then this `if` is true? ie, the `diff` is not empty, but `addedInsertions`  is? 
would that be a bug in EM?

Reply:
the difference might be due to db action repair. 
here, we aim to keep consistency between genes and their impacts.

Reply:
Hi, @arcuri82 
I cannot comment on the code which has been merged.
Please check the line 777,

``val diff = individual.seeInitializingActions()
            .filter { !old.contains(it) && ((it is SqlAction && !it.representExistingData) || it is MongoDbAction) }``

I am not sure whether the `MongoDbAction` should be always considered as `diff`. 
Could you please have a check?